### PR TITLE
MC Tidy

### PR DESCRIPTION
**Pull request by trisyoungs**
This PR started off as a bug hunt attempting to fix the `MonteCarloMinimiser` class which no longer appeared to modify any values, and ended up as a complete simplification of said class, removing redundant functionality and improving the basic method.

### CODE REVIEW

**Review by IVlaD17 — APPROVED** [review]
I can't comment too much on the scientific side of this, but, in terms of the code itself, it looks clean enough for me :)

File: src/math/poissonfit.cpp [hunk 1]
```diff
@@ -436,28 +445,44 @@ double PoissonFit::constructReciprocal(double rMin, double rMax, const std::vect
...
+
+            // Add in contributions from our Gaussians
+            for (auto n = 0; n < C_.size(); ++n)
+                y += functions_[{n, i}] * C_[n];
+
+            dy = referenceData_.value(i) - y;
+            sose += dy * dy;
+        }
+
+        return sose;
```
Comment (rprospero) [inline src/math/poissonfit.cpp:472]: This lambda appears twice.  Maybe we should make it a member function and just pass is into the minimiser?

**Review by rprospero — APPROVED** [review]
I'd tried to work on doing some of this clean-up, but this is much nicer than what I had tried.  Besides one place where I think we can factor a repeated lambda out into a single function, I think that this is good to go in.

File: src/math/poissonfit.cpp (hunk 1 above)
Comment (trisyoungs) [reply src/math/poissonfit.cpp]: :+1:

**Pull request merged**